rfd/0106-machine-id-anonymous-telemetry.md

```diff
@@ -0,0 +1,156 @@
+
+service TbotReportingService {
+  rpc SubmitTbotEvent(SubmitTbotEventRequest) returns (SubmitTbotEventResponse) {}
+}
+```
+
+As only a single event will be submitted, batching is not of concern.
+
+### Event storage
+
+Events received by `prehog` will be encoded in a PostHog compatible format
+and then submitted to PostHog.
+
+Tbot events will be share the same project as clusters and website events.
+
+### Consent
+
+Users will explicitly opt-in to anonymous usage telemetry with an environment
+variable, e.g:
+
+```sh
+# Send anonymous telemetry on startup about the usage of tbot in order
+# to help us understand what parts of Machine ID to improve.
+# Find out more at goteleport.com/awesome-docs-page
+export TBOT_ANONYMOUS_TELEMETRY=1
+tbot start --snip--
+```
+
+For helpers like GitHub actions, this will be another option the user can
+provide, e.g:
+
+```yaml
+steps:
+  - name: Install Teleport
+    uses: teleport-actions/setup@v1
+    with:
+      version: 11.0.3
+  - name: Authorize against Teleport
+    id: auth
+    uses: teleport-actions/auth@v1
+    with:
+      # Send anonymous telemetry on startup about the usage of tbot in order
+      # to help us understand what parts of Machine ID to improve.
+      # Find out more at goteleport.com/awesome-docs-page
+      anonymous-telemetry: true
+```
+
+Examples provided in the documentation will be updated to include this value
+to encourage opt-in, but will make clear that this parameter can be removed.
+
+A documentation page will be created that will explain what telemetry is
+collected and where this telemetry is stored.
+
+When telemetry is enabled, a log message will be output on startup stating that
+telemetry is being collected and linking to the relevant documentation.
+
+When telemetry is not enabled, a log message will be output on startup
+informing the user that Telemetry is not enabled and linking to the relevant
+documentation.
+
+### Anonymization
+
+The events will contain no properties that identify a user, `tbot` or Teleport
+cluster. Therefore, no additional anonymization is required at this time.
+
+### Events
+
+#### `tbot.start`
+
+Event properties:
+
+- `tbot.run_mode`: one of [`one-shot`, `daemon`]
+- `tbot.version`: string indicating the version of `tbot`
+- `tbot.join_type`: string indicating the join type configured
+- `tbot.helper`: optional string indicating if a helper is invoking `tbot`. For 
+  example: `gha:teleport-actions/auth`
+- `tbot.helper_version`: optional string indicating the version of the helper 
+  that invoking `tbot`
+- `tbot.destinations_configured`: a count of total destinations configured
+- `tbot.destinations_configured_database_access`: a count of Database Access 
+  destinations configured
+- `tbot.destinations_configured_kubernetes_access`: a count of Kubernetes Access
+  destinations configured
+- `tbot.destinations_configured_application_access`: a count of Application 
+  Access destinations configured.
```

Comment:
Why is the sum a separate property? For analytics purposes?

Reply:
Perhaps `destinations_configured` should have a different name, maybe with `_other` or something?

Reply:
Good catch - this could be clearer and more concise:

f13d3b5409